core/shadow-selection-polyfill.js

```diff
@@ -4,7 +4,7 @@ const SUPPORTS_BEFORE_INPUT = typeof window.InputEvent.prototype.getTargetRanges
 const IS_FIREFOX = window.navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
```

Comment:
Do we need to handle FF differently ?

Reply:
Differently than Safari? Yes, it does not need to use the ShadowSelection class. It can do everything using `document.getSelection()`.